arch/x86/crc32_fold_pclmulqdq_tpl.h

```diff
@@ -105,6 +105,205 @@ Z_INTERNAL void CRC32_FOLD(crc32_fold *crc, const uint8_t *src, size_t len, uint
     }
 #endif
 
+
+        while (len >= 512 + 64 + 16*8) {
+                __m128i shift544_shift480 = _mm_set_epi64x(0x1D9513D7, 0x8F352D95);
+                // __m128i shift672_shift608 = _mm_set_epi64x(0xAE0B5394, 0x1C279815);
+                // __m128i shift800_shift736 = _mm_set_epi64x(0x57C54819, 0xDF068DC2);
+                // __m128i shift1568_shift1504 = _mm_set_epi64x(0x910EEEC1, 0x33FFF533);
+                __m128i shift1568_shift1504 = _mm_set_epi64x(0xF5E48C85, 0x596C8D81);
+            __m128i bonus8 = _mm_loadu_si128(src + (16 * 4));
+            __m128i bonus7 = _mm_loadu_si128(src + (16 * 5));
+            __m128i bonus6 = _mm_loadu_si128(src + (16 * 6));
+            __m128i bonus5 = _mm_loadu_si128(src + (16 * 7));
+            __m128i bonus4 = _mm_loadu_si128(src + (16 * 8));
+            __m128i bonus3 = _mm_loadu_si128(src + (16 * 9));
+            __m128i bonus2 = _mm_loadu_si128(src + (16 * 10)) ^ bonus8;
+            __m128i bonus1 = _mm_loadu_si128(src + (16 * 11)) ^ bonus7;
+            src += 16*8;
+            len -= 16*8;
+            xmm_t0 = _mm_loadu_si128(src + (16 * 4)) ^ bonus6;
+            xmm_t1 = _mm_loadu_si128(src + (16 * 5)) ^ bonus5 ^ bonus8;
+            xmm_t2 = _mm_loadu_si128(src + (16 * 6)) ^ bonus4 ^ bonus8 ^ bonus7;
+            xmm_t3 = _mm_loadu_si128(src + (16 * 7)) ^ bonus3 ^ bonus7 ^ bonus6;
```

Comment:
There's a high probability those XORs won't compile for MSVC, I think only GCC and clang support binary operations on SIMD operands.  With AVX512 there's also an opportunity here to use ternary logic here, though we'll need to pollute things with an ifdef macro like we do in some other places.

Reply:
That's good to know, I did the binary operations to make the code clearer but happy to rewrite with intrinsics

Reply:
New commit replaces xors with `_mm_xor_si128()` intrinsic